content/cybersecurity/concepts/sql-injection/sql-injection.md

```diff
@@ -0,0 +1,54 @@
+---
+Title: 'SQL Injection'
+Description: 'SQL injection is a technique used to inject malicious code into a database.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Database'
+  - 'Queries'
+  - 'SQLite'
+  - 'MySQL'
+  - 'PostgreSQL'
+CatalogContent:
+  - 'learn-sql'
+  - 'paths/analyze-data-with-sql'
+---
+
+**SQL injection** is a technique used to introduce unwanted code into a [SQL](https://www.codecademy.com/resources/docs/sql) database. This code is malicious and can be sent to a database and run without permission. The result can be anything from bypassing login to data theft or corruption of a database. The code is usually entered when a website or application asks for user input.
+
+To prevent SQL injection, databases should not take user input as a command to be run. Parameters can be used to protect against SQL injection, allowing for more control over the values being passed. This will ensure SQL will treat the data as text and not as a command to be executed.
+
+## Vulnerable Queries
+
+In this example, there is a form that asks for a `User ID`. This form will run this SQL command to check if the user exists.
+
+```sql
+SELECT * FROM users WHERE UserID = 'input from user';
+```
+
+Now if a user enters `1' OR '1'='1`, then the command will look like the following:
```

Comment:
```suggestion
Now if a user enters `'1' OR '1'='1'`, then the command will look like the following:
```

Reply:
This looks odd to me. Should this include single quotes as I've added above? 

The same issue occurs on line 38

Reply:
@caupolicandiaz No, the outside quotes are provided by the query, since it's placing the user input inside single quotes.